src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoParticipantDataDao.java

```diff
@@ -0,0 +1,130 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.PaginatedQueryList;
+import com.amazonaws.services.dynamodbv2.datamodeling.QueryResultPage;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.dao.ParticipantDataDao;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.PagedResourceList;
+import org.sagebionetworks.bridge.models.ParticipantData;
+import org.sagebionetworks.bridge.models.ResourceList;
+import org.sagebionetworks.bridge.models.reports.ReportData;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+@Component
+public class DynamoParticipantDataDao implements ParticipantDataDao {
+    private DynamoDBMapper mapper;
+
+    @Resource(name = "participantDataMapper")
+    final void setParticipantStateMapper(DynamoDBMapper participantStateMapper) {
+        this.mapper = participantStateMapper;
+    }
+
+    @Override
+    public ForwardCursorPagedResourceList<? extends ParticipantData> getParticipantData(String userId, String offsetKey, int pageSize) {
+        checkNotNull(userId);
+
+        DynamoParticipantData hashKey = new DynamoParticipantData();
+        hashKey.setUserId(userId);
+
+        DynamoDBQueryExpression<DynamoParticipantData> query =
+                new DynamoDBQueryExpression<DynamoParticipantData>().withHashKeyValues(hashKey);
+        List<DynamoParticipantData> results = mapper.query(DynamoParticipantData.class, query);
+
+        return new ForwardCursorPagedResourceList<DynamoParticipantData>(results, offsetKey);
+    }
+
+    @Override
+    public ForwardCursorPagedResourceList<ParticipantData> getParticipantDataV4(final String userId, final String offsetKey, final int pageSize) {
+        //TODO: why do we use final for these params but not the others?
+        checkNotNull(userId);
+
+        int pageSizeWithIndicatorRecord = pageSize + 1; //TODO: what is indicatorRecord?
```

Comment:
Let's say you retrieve 10 records. You want to return the ID of the 11th record to indicate where the next page should begin, so we get page + 1 records to get that 11th record ID.